pkg/controller/che/che_controller.go

```diff
@@ -842,19 +841,26 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		return reconcile.Result{}, err
 	}
 
-	// create a custom ConfigMap that won't be synced with CR spec
-	// to be able to override envs and not clutter CR spec with fields which are too numerous
-	customCM := &corev1.ConfigMap{
-		Data: deploy.GetCustomConfigMapData(),
-		TypeMeta: metav1.TypeMeta{
-			Kind: "ConfigMap"},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "custom",
-			Namespace: instance.Namespace,
-			Labels:    cheLabels}}
-	if err := r.CreateNewConfigMap(instance, customCM); err != nil {
-		return reconcile.Result{}, err
+	// Get custom ConfigMap
+	// if it exists, add the data into OverrideCheProperties
+	customConfigMap := &corev1.ConfigMap{}
```

Comment:
I would move this at the beginning of the reconcile method, to line https://github.com/eclipse/che-operator/pull/87/files#diff-d9445e614f0d728fd140771b0fd29c50L243

So it would be the first thing done.

Reply:
Do you want me to just move the declaration of the config map, or the whole reconciliation logic piece?

Reply:
I was meaning move this whole block https://github.com/eclipse/che-operator/pull/87/files#diff-d9445e614f0d728fd140771b0fd29c50R846-R863 :

The maintenance of the `custom` config map.